frontend/src/services/UserFunctions.ts

```diff
@@ -0,0 +1,40 @@
+import { UserCredential, getAuth, onAuthStateChanged } from "firebase/auth"; // Import UserCredential type
+import { SuccessObject, callUserFunction } from "@/lib/utils";
+
+export async function addToUserCollection(userCredential: UserCredential): Promise<SuccessObject> {
+    const { uid, email, displayName } = userCredential.user;
+    // Prepare the data to send in the request body
+    const userData = {
+        uid,
+        email,
+        displayName: displayName || '', // Use an empty string if displayName is not provided
+    };
+    console.log("sending user to collection...")
+    const res = await callUserFunction("user/addToUserCollection", "POST", userData);
```

Comment:
is the admin flag defaulted to false ?

Reply:
yup I set it to false in firebase. I chose not to leave a field in here so that there is no way for the frontend to send isAdmin true.